mongo/integration/cmd_monitoring_helpers_test.go

```diff
@@ -180,6 +181,16 @@ func compareDocsHelper(mt *mtest.T, expected, actual bson.Raw, prefix string) er
 		}
 
 		aVal, err := actual.LookupErr(eKey)
+		if e.Value().Type == bson.TypeNull {
+			// Expected value is BSON null. Expect the actual field to be omitted.
+			if err == bsoncore.ErrElementNotFound {
+				continue
+			} else if err != nil {
+				return fmt.Errorf("expected key %q to be omitted but got error: %v", eKey, err)
+			} else {
+				return fmt.Errorf("expected key %q to be omitted but got %q", eKey, aVal)
+			}
+		}
```

Comment:
```suggestion
			// Expected value is BSON null. Expect the actual field to be omitted.
			if errors.Is(err, bsoncore.ErrElementNotFound) {
				continue
			}

			if err != nil {
				return fmt.Errorf("expected key %q to be omitted but got error: %v", eKey, err)
			}

			return fmt.Errorf("expected key %q to be omitted but got %q", eKey, aVal)
```

This suggestion:

(1) removes the if/else branching
(2) uses the `errors` package to check if the error is `bsoncore.ErrElementNotFound`